src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -74,4 +109,49 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
     protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest request) {
         return AuthorisationUtil.getTokenPermissions(request);
     }
+
+    protected boolean isRequestROEUpdate(HttpServletRequest request, String transactionId) {
+        final Map<String, List<String>> privileges = getERICTokenPermissions(request);
+        String companyInScope;
+        if (privileges.containsKey(COMPANY_NUMBER_KEY)) {
+            companyInScope = privileges.get(COMPANY_NUMBER_KEY).get(0);
+            if (StringUtils.isNotEmpty(companyInScope) && companyInScope.startsWith("OE")) {
+                String companyInTransaction = getCompanyInTransaction(request, transactionId);
```

Comment:
I don't think this part should be in this method - you're on the ROE update journey if the company number key is populated with a number starting with 'OE'. If it does pass that check but company number doesn't match that against the transaction then that is something else.

I suggest extracting the logic that gets the company number in the transaction and checking it against the number in the header  and calling this on line 87 with an AND clause.